src/modules/attempts/attempts.service.ts

```diff
@@ -305,8 +305,117 @@ export class AttemptsService {
       throw new NotFoundException('Attempt not found');
     }
 
+    // Validate attempt status
+    if (attempt.status !== AttemptStatus.SUBMITTED) {
+      throw new BadRequestException('Only submitted attempts can be reviewed');
+    }
+
+    // Validate review status
+    if (attempt.reviewStatus !== ReviewStatus.PENDING) {
```

Comment:
there can be a case where, reviewer has opened the answersheet for review which was having status Pending. Now he reviews 4 answers out of 6, so then the attempt review status is under review right? @vinayak-patil